src/types/Login.ts

```diff
@@ -0,0 +1,29 @@
+export interface LoginProvider {
+  method: string,
+  button: string,
+  text: string,
+}
+
+export interface LoginRedirect {
+  redirect: string
+}
+
+// Corresponds to body returned by `/auth/userinfo`
+export interface UserInfo {
+  displayname: string,
+  github_login: string,
+  github_avatar: string,
+  'dev-flatpaks': string[],
+}
```

Comment:
Only corresponds currently, a TODO for this to be re-addressed when we have other login kinds will likely be necessary.